shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -43,8 +46,27 @@ internal class StepQuizActionDispatcher(
     private val sentryInteractor: SentryInteractor,
     private val onboardingInteractor: OnboardingInteractor,
     private val resourceProvider: ResourceProvider,
-    private val platform: Platform
+    private val platform: Platform,
+    currentSubscriptionStateRepository: CurrentSubscriptionStateRepository
 ) : CoroutineActionDispatcher<Action, Message>(config.createConfig()) {
+
+    init {
+        currentSubscriptionStateRepository
+            .changes
+            .onEach { subscription ->
+                onNewMessage(
+                    InternalMessage.ProblemsLimitChanged(
+                        isProblemsLimitReached = subscription.isProblemsLimitReached,
+                        problemsLimitReachedModalData = getProblemsLimitReachedModalData(
+                            profile = currentProfileStateRepository.getState().getOrElse { return@onEach },
+                            subscription
+                        )
+                    )
+                )
+            }
+            .launchIn(actionScope)
```

Comment:
This implementation conflicts with `freemium.wrong_submission_charge_limits` feature, see https://github.com/hyperskill/mobile-app/pull/921

When problems limits reached, modal will be presented twice, maybe something else...

It was supposed that:
1. `StepQuizFeature.InternalMessage.UpdateProblemsLimitResult`
2. `StepQuizFeature.InternalAction.UpdateProblemsLimit`

will be used only when `freemium.wrong_submission_charge_limits` is enabled